lib/elixir/lib/json.ex

```diff
@@ -150,6 +150,12 @@ defimpl JSON.Encoder, for: Map do
   end
 end
 
+defimpl JSON.Encoder, for: [Date, Time, NaiveDateTime, DateTime] do
```

Comment:
Should `Duration` also be included?

Reply:
I asked myself the same and, honestly,  I am not sure. We could easily use the ISO representation, but I am not sure how widespread it is, it is definitely less common than the datetime ones.

Reply:
AFAIK, in the JS world it is common to serialise duration to ISO strings when parsing it to JSON (see [`Temporal.Duration.toJSON`](https://tc39.es/proposal-temporal/docs/duration.html#toJSON) or [`dayjs.duration`](https://day.js.org/docs/en/durations/as-json)). As the string format chose in serialization is just a convention, I hopped ISO strings was a good enough format to be used as default.